nil/services/rpc/jsonrpc/txpool_api_test.go

```diff
@@ -0,0 +1,106 @@
+package jsonrpc
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/NilFoundation/nil/nil/common/logging"
+	"github.com/NilFoundation/nil/nil/internal/db"
+	"github.com/NilFoundation/nil/nil/internal/types"
+	"github.com/NilFoundation/nil/nil/services/rpc/rawapi"
+	"github.com/NilFoundation/nil/nil/services/txnpool"
+	"github.com/stretchr/testify/suite"
+)
+
+type SuiteTxnPoolApi struct {
+	SuiteAccountsBase
+	txnpoolApi *TxPoolAPIImpl
+	api        *rawapi.NodeApiOverShardApis
+	pool       txnpool.Pool
+}
+
+const defaultMaxFee = 500
+
+func newTransaction(address types.Address, seqno types.Seqno, priorityFee uint64, code types.Code) *types.Transaction {
+	return &types.Transaction{
+		TransactionDigest: types.TransactionDigest{
+			To:                   address,
+			Seqno:                seqno,
+			MaxPriorityFeePerGas: types.NewValueFromUint64(priorityFee),
+			MaxFeePerGas:         types.NewValueFromUint64(defaultMaxFee),
+			Data:                 code,
+		},
+	}
+}
+
+func (suite *SuiteTxnPoolApi) SetupSuite() {
+	suite.SuiteAccountsBase.SetupSuite()
+	var err error
+	ctx := context.Background()
+
+	suite.pool, err = txnpool.New(ctx, txnpool.NewConfig(types.MainShardId), nil)
+	suite.Require().NoError(err)
+
+	database, err := db.NewBadgerDbInMemory()
+	suite.Require().NoError(err)
+	defer database.Close()
+
+	localShardApis := map[types.ShardId]rawapi.ShardApi{
+		types.MainShardId: rawapi.NewLocalShardApi(types.MainShardId, database, suite.pool),
+	}
+
+	shardApis := make(map[types.ShardId]rawapi.ShardApi)
+	localShardApi := rawapi.NewLocalShardApi(types.MainShardId, database, suite.pool)
+
+	shardApis[types.MainShardId], err = rawapi.NewLocalRawApiAccessor(types.MainShardId, localShardApi)
+
+	localApi := rawapi.NewNodeApiOverShardApis(localShardApis)
+	suite.txnpoolApi = NewTxPoolAPI(localApi, logging.NewLogger("Test"))
+	suite.api = rawapi.NewNodeApiOverShardApis(shardApis)
```

Comment:
I don't understand why we need `localShardApis` and `shardApis` at the same time. Why don't we want to do just that?
```go
	mainShardApi,err := rawapi.NewLocalRawApiAccessor(
		types.MainShardId,
		rawapi.NewLocalShardApi(types.MainShardId, database, suite.pool))
	suite.Require().NoError(err)
	localShardApis := map[types.ShardId]rawapi.ShardApi{types.MainShardId: mainShardApi}

	suite.txnpoolApi = NewTxPoolAPI(localShardApis, logging.NewLogger("Test"))
	suite.api = rawapi.NewNodeApiOverShardApis(localShardApis)
```